src/ziggurat/mapper.clj

```diff
@@ -52,6 +53,8 @@
                 :success (metrics/multi-ns-increment-count multi-message-processing-namespaces success-metric additional-tags)
                 :retry (do (metrics/multi-ns-increment-count multi-message-processing-namespaces retry-metric additional-tags)
                            (producer/retry message-payload))
+                :dead-letter (do (metrics/multi-ns-increment-count multi-message-processing-namespaces dead-letter-metric additional-tags)
```

Comment:
I would recommend using `:fail-now` or `:fail` or `:failure` or some status like that to signal failure instead of `:dead-letter`
(this is more generic since it agnostic of queue related terminology and also in-case RMQ is not  enabled, we might just want to mark failure and move on)

Reply:
`:failure` sounds good to me. In this case so we increment `failure-metric`? Because the same metric is being used to for exceptions

```clojure
(catch Throwable e
              (producer/retry message-payload)
              (report-error e (str "Actor execution failed for " topic-entity-name))
              (metrics/multi-ns-increment-count multi-message-processing-namespaces failure-metric additional-tags))))))))
```

Reply:
I think dead-letter is pretty standard term in stream processing. 

https://en.wikipedia.org/wiki/Dead_letter_queue

Rest all looks good to me. @clouddra Have you tested the change locally too?

Reply:
Unfortunately I am not too familiar with clojure (was intending to using this via a java client). The only testing i did was the unit test that I have added. If that is insufficient any advice as to how I may test it?

Reply:
I managed to test it with a java client. Works as expected. Messages get pushed into DLQ immediately without retrying and skipping the delay queues.

Reply:
@mjayprateek Yes, but moving to DLQ is not the intention here 🤦 . The intention of returning this special status is to tell Ziggurat not to retry and fail immediately. Whether it moves to a DLQ or simply gets discarded is an after-effect.

Hence was suggesting to use a generic terminology.